TimeSignal time_signal
CustomSignal custom_signal
---
string response